### PR TITLE
Initial documentation for Multi-cluster support.

### PR DESCRIPTION
**Pull request by absoludity**
### Description of the change

Provides initial documentation for the multi-cluster support.

View [the rendered markdown](https://github.com/absoludity/kubeapps/blob/1764-document-multicluster/docs/user/deploying-to-multiple-clusters.md)

### Applicable issues

  - Ref #1764

### CODE REVIEW

**Review by andresmgot — APPROVED** [review]
IMO, this guide exposes generic problems but I was expecting a step by step guide to have this working, even if it's just for a kind + dex demo environment. Speaking for myself, it wouldn't be trivial for me to setup this environment following this.

I am approving to not block the release, I believe the guide I mention is not needed for people to start trying this. Thanks!

File: docs/user/deploying-to-multiple-clusters.md [hunk 1]
```diff
@@ -0,0 +1,83 @@
...
+
+Alternatively, for a development with private API server URLs, you can omit the `certificateAuthorityData` and instead include the field `insecure: true` for a cluster and Kubeapps will not try to verify the secure connection.
+
+A serviceToken is not required but provides a better user experience, enabling users viewing the cluster to see the namespaces to which they have access (only) when they use the namespace selector. The service token should be configured with RBAC so that it can only list namespaces on the cluster. You can refer to the [example used for a local development environment](/docs/user/manifests/kubeapps-local-dev-namespace-discovery-rbac.yaml).
+
+Your Kubeapps installation will also need to be [configured to use OIDC for authentication](/docs/user/using-an-OIDC-provider.md) with a client-id for your chosen provider.
+
+## Clusters with different client-ids
+
+Some multi-cluster environments configure each cluster's API server with its own client-id for the chosen OAuth2 provider. In this case, there is some extra configuration required to ensure the OIDC token used by Kubeapps is accepted by the different clusters.
```
[the remark this answers is not in the document]
Comment (absoludity) [reply docs/user/deploying-to-multiple-clusters.md]: The next two sub-sections have examples of doing this with dex. I've updated the above sentence to show an example from the TKG documentation where you are asked to do this, and ensured that it's obvious that the following sub-sections flow from here.

**Review by absoludity — COMMENTED** [review]
> but I was expecting a step by step guide to have this working, even if it's just for a kind + dex demo environment. Speaking for myself, it wouldn't be trivial for me to setup this environment following this.

Yes - it's not trivial to setup multiple K8s clusters with oidc support. Read the TKG instructions and you'll see it involves *many* pages of specific instructions to edit certain files with certain values etc. Luckily we've automated it for a local dev environment.

It also wasn't trivial for me to *find* a solution to the multiple client-ids used by TKG (spent a good part of a week looking for the best solution which turns out to be relatively trivial - trusted peers and requesting extra audiences via the scope).

Regarding a step-by-step setup for kind + dex, I wanted to mention that it's automated (ie. if you have kind and mkcerts installed, you can just `make multi-cluster-kind` to get the two clusters setup ready to trust dex, and then after exporting your KUBECONFIG, `make deploy-dev` to install dex, openldap and kubeapps all working together in a multi-cluster env, but I haven't included this in the doc. Perhaps we should?

> 
> I am approving to not block the release, I believe the guide I mention is not needed for people to start trying this. Thanks!

Great, thanks.

File: docs/user/deploying-to-multiple-clusters.md [hunk 2]
```diff
@@ -0,0 +1,105 @@
...
+
+### Configuring the auth-proxy to request multiple audiences
+
+The second part of the additional configuration is to ensure that when Kubeapps' auth-proxy requests a token that it includes extra scopes, such as `audience:server:client_id:second-cluster` for each additional audience that it requires in the issued token. For example, you can view the [auth-proxy configuration used in the local development environment](/docs/user/manifests/kubeapps-local-dev-auth-proxy-values.yaml) and see the additional scopes included there to ensure that the `second-cluster` and `third-cluster` are included in the audience of the resulting token.
+
+## Updating multicluster options
+
+TBD
```
Comment (andresmgot) [inline docs/user/deploying-to-multiple-clusters.md:95]: BTW, this `TBD` seems like the documentation is not finished (while I believe there is an open issue to support this), maybe we should elaborate this a bit more?

**Review by andresmgot — COMMENTED** [review]
> Regarding a step-by-step setup for kind + dex, I wanted to mention that it's automated (ie. if you have kind and mkcerts installed, you can just `make multi-cluster-kind` to get the two clusters setup ready to trust dex, and then after exporting your KUBECONFIG, `make deploy-dev` to install dex, openldap and kubeapps all working together in a multi-cluster env, but I haven't included this in the doc. Perhaps we should?

Yes please, at the very minimum, it will be helpful for me to set this up quickly in my dev environment.

File: docs/user/deploying-to-multiple-clusters.md (hunk 2 above)
Comment (absoludity) [reply docs/user/deploying-to-multiple-clusters.md:95]: It was intended to be understood that the documentation wasn't finished - I've filled this in with #1955 . Not sure which open issue you're thinking of?

**Comment by absoludity** [comment]
> Yes please, at the very minimum, it will be helpful for me to set this up quickly in my dev environment.

Also done in #1955 .

File: docs/user/deploying-to-multiple-clusters.md (hunk 2 above)
Comment (andresmgot) [reply docs/user/deploying-to-multiple-clusters.md:95]: ah, I thought this was related to https://github.com/kubeapps/kubeapps/issues/1942

**Pull request merged**